src/main/java/database/DataBaseService.java

```diff
@@ -1,63 +1,48 @@
 package database;
 import java.sql.Connection;
-import java.sql.DriverManager;
 import java.sql.PreparedStatement;
 import java.sql.SQLException;
 
-import controller.ConfigurationService;
 import pojo.Switch;
 import pojo.Temperature;
 
 public class DataBaseService {
 	
-	private ConfigurationService configurationService;
+	private DataSource dataSource;
 	private Connection conn;
```

Comment:
Neizmanto konekciju kā instances mainīgo, jo tagad Tev `conn` reference tiek pārrakstīta visās vietās, kur izsauc: `conn = dataSource.getConnection();`, kas nozīmē, ka tad, kad Tev būs paralēli pieprasījumi, tad izveidosies dīvaina situācija, ka sql sāks izpildīt ar vienu konekciju, bet tad, kad tiks līdz konekcijas aizvēršanai, tad aizvērs (mēģinās aizvērt) citu, jo pa to laiku, jau būs izpildījies kāds izsaukums, kas pārraksta `conn` referenci.